google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java

```diff
@@ -866,6 +975,71 @@ private void cleanupInflightRequests() {
     }
   }
 
+  private Boolean retryOnRetryableError(Code errorCode, AppendRequestAndResponse requestWrapper) {
+    if (this.maxRetryNumAttempts == 0) {
+      return false;
+    }
+
+    if (!isImmediatelyRetriableStatus(errorCode) && errorCode != Code.RESOURCE_EXHAUSTED) {
+      return false;
+    }
+
+    if (requestWrapper.retryCount < this.maxRetryNumAttempts) {
+      lock.lock();
+      try {
+        requestWrapper.retryCount++;
+        if (this.retrySettings != null && errorCode == Code.RESOURCE_EXHAUSTED) {
+          // Trigger exponential backoff in append loop when request is resent for quota errors
+          if (requestWrapper.attemptSettings == null) {
+            requestWrapper.attemptSettings = requestWrapper.retryAlgorithm.createFirstAttempt();
+          } else {
+            requestWrapper.attemptSettings =
+                requestWrapper.retryAlgorithm.createNextAttempt(requestWrapper.attemptSettings);
+          }
+          requestWrapper.blockMessageSendDeadline =
+              Instant.now().plusMillis(requestWrapper.attemptSettings.getRetryDelay().toMillis());
+        }
+
+        Long offset =
+            requestWrapper.message.hasOffset() ? requestWrapper.message.getOffset().getValue() : -1;
+        if (isDefaultStreamName(streamName) || offset == -1) {
+          log.fine(
+              String.format(
+                  "Retrying default stream message in stream %s for in-stream error: %s, retry count:"
+                      + " %s",
+                  streamName, errorCode, requestWrapper.retryCount));
+          addMessageToFrontOfWaitingQueue(requestWrapper);
+        } else {
+          log.fine(
+              String.format(
+                  "Retrying exclusive message in stream %s at offset %d for in-stream error: %s, retry"
+                      + " count: %s",
+                  streamName,
+                  requestWrapper.message.getOffset().getValue(),
```

Comment:
Right now those retries are being done in a quite transparent manner, can we at least add some periodical log that's show some retry is happening?

Reply:
Wouldn't line 1015 do what you are asking?  Or maybe change that log to info to make it more visible?

Reply:
hmm, info might flood the log too much, my experience is almost no one ever debugging using the .fine() log level because there's just too much of it to parse information.

Let's at least change the one "Max retry count reached for message..." to info

Reply:
Done